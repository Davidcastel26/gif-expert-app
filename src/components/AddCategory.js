import {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setinputValue(e.target.value)

        console.log('handleinputchange called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        // console.log('sub done');
        // we are calling categories

        console.log('hangleSumit', inputValue);

        if( inputValue.trim().length > 2){
            setCategories( categs => [ inputValue, ...categs ] )
            setinputValue('')
        }

    } 

    return (
        <form onSubmit={ handleSubmit }> 
            <p> { inputValue} </p> 
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;