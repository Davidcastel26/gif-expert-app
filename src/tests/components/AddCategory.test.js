import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import React from 'react';

describe('testing the component add category', () => {

    const setCategories = () => {};

    test('should show the right path', () =>{

        const wrapper = shallow( <AddCategory setCategories={setCategories} />)

        expect( wrapper ).toMatchSnapshot();

    })

})