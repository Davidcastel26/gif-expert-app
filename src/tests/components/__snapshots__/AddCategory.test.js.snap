// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component add category should show the right path 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
